lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -1727,18 +1725,78 @@ class ConvertAtenScaledDotProductAttentionOp
       mask = genericOp.getResult(0);
     }
 
+    // Broadcast the batch dimensions of the mask:
+    if (mask) {
+      auto maskTy = cast<RankedTensorType>(mask.getType());
+      int64_t rank = maskTy.getRank();
+      bool needsBroadcast = false;
+      for (int i = 0, s = rank - 2; i < s; ++i) {
+        needsBroadcast |= maskTy.getDimSize(i) != keyTy.getDimSize(i);
```

Comment:
did not know there was an or= operator. thats cool

Reply:
You can go weirder with `^=` for xor equals